### ISSUE TITLE
WKT cannot load geocollections if one of the shapes is "EMPTY"

### ISSUE THREAD

**Issue by cjshowalter** [geomet/geomet#49]
### What is expected:
These shapes load up great:
```python
geomet.wkt.loads("GEOMETRYCOLLECTION EMPTY")
geomet.wkt.loads("POINT EMPTY")
```
### What happens:
It's when one of the shapes *inside* the geocollection contain an empty shape, I get a ValueError:
```python
geomet.wkt.loads("GEOMETRYCOLLECTION (POLYGON((27 25,102 36,102 46,92 61,13 41,16 30,27 25)), LINESTRING EMPTY )")
```
Should it be able to create a json like so?
```json
{'type': 'GeometryCollection', 'geometries': [
    {'type': 'Polygon', 'coordinates': [[[27.0, 25.0], [102.0, 36.0], [102.0, 46.0], [92.0, 61.0], [13.0, 41.0], [16.0, 30.0], [27.0, 25.0]]]},
    {'type': 'LineString', 'coordinates': []}
]}

```
### Maybe of note:
When I try to load the above geocollection w/ the empty shape inside it to [shapely](https://pypi.org/project/Shapely/), then dump it, I get this:
```python
GEOMETRYCOLLECTION Z (POLYGON((27 25,102 36,102 46,92 61,13 41,16 30,27 25)), LINESTRING EMPTY )
```
with the "geometrycollection z". I'm new to wkt's so I'm not sure if the Z's expected as well, just thought I'd mention it.
Thanks!

**Comment by larsbutler** [comment]
@cjshowalter Hmm, that is indeed a bug. Good catch. It seems I implemented the "EMPTY" handling in the wrong place, creating an edge case.

Basically, [this logic](https://github.com/geomet/geomet/blob/60b02ef7596c318cc06b2cf8fe9d2d80372e3046/geomet/wkt.py#L137-L143) needs to move down into the individual `_load_*` functions. I suppose I was trying to be clever by abstracting it into the top-level `loads` function, but that's broken.

> When I try to load the above geocollection w/ the empty shape inside it to shapely, then dump it, I get this ... with the "geometrycollection z". I'm new to wkt's so I'm not sure if the Z's expected as well, just thought I'd mention it.

That's definitely weird that it's being read as a Z geometry. Z-type geometries (see https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry) should always have 3 dimensions, so it seems strange that shapely is treating it like that.

Quick tangent: All of the different formats' specifications (GeoJSON, WKT, and WKB) don't really have a consistent way of deal with "empty" geometries and geometries of different dimensions. For example, in GeoJSON, there's no way to tell if a 3-dimensional geometry is xyz or xym. As for "empty" geometries, the entire concept is flawed if you ask me. For collection-type geometries it makes sense: empty geometry collections, empty multipoints, etc. But what is an empty "point"? In pure geometric terms, how can a point be "empty" (or "full", for that matter) when it is simply a location with [no width, length, or breadth](https://mathcs.clarku.edu/~djoyce/java/elements/bookI/defI1.html)? My suspicion is that all of this confusion originated from the [Java Topology Suite](https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/geom/Point.java#L29-L30) and the way is defines `isEmpty` as a method of every geometry type. This is a premature abstraction that survived a C++ port of the JTS called [libGEOS](https://github.com/libgeos/geos). libGEOS underpins basically every major geospatial software today, including PostGIS. That's a scary little piece of tech debt which leads to some interesting undefined edge cases.

**Comment by cjshowalter** [comment]
@larsbutler wow, thanks for the fast reply!

> For example, in GeoJSON, there's no way to tell if a 3-dimensional geometry is xyz or xym.

Just with a list of coords, I can't think of a way to do this either. I wouldn't be surprised if that's why they added the tags to the end of the name itself. (ie *POINT ZM (...)*). That'd probably require a huge code rewrite to implement though xD

> What is an empty "point"? In pure geometric terms, how can a point be "empty" (or "full", for that matter) when it is simply a location with [no width, length, or breadth](https://mathcs.clarku.edu/~djoyce/java/elements/bookI/defI1.html)? 

This might be a over simplification, but I see a empty point as *not* having a location. I see it being used kinda like a "Null" value of shapes in that regard. In my case, I can just ignore them and grab the non-empty shapes, but maybe if you needed to delete overlapping points from different shapes, one of them could become empty in the process.

**Comment by tomplex** [comment]
@cjshowalter This issue is resolved in #53 which I just merged. Not sure when we'll get a new version of `geomet` out but I'll update this issue again when that happens. Thanks again for the bug report!

**Comment by tomplex** [comment]
@cjshowalter This is resolved with the release of `geomet` 0.3.0.
